framework/include/actioncomponents/ComponentBoundaryConditionInterface.h

```diff
@@ -0,0 +1,98 @@
+{
+public:
+  static InputParameters validParams();
+
+  ComponentBoundaryConditionInterface(const InputParameters & params);
+
+  enum BoundaryConditionType
+  {
+    FIXED_VALUE,
+    FLUX
+  };
+
+  /**
+   * Whether the component has a boundary condition parameter is specified for the requested
+   * variable
+   * @param variable the name of the variable
+   * @return Whether the component has a boundary condition parameter specified for the requested
+   * variable
+   */
+  bool hasBoundaryCondition(const VariableName & variable) const;
+  /**
+   * Whether the component has a boundary condition parameter is set for the requested variable and
+   * boundary
+   * @param variable the name of the variable
+   * @param boundary the name of the boundary
+   * @return Whether the component has a boundary condition parameter specified
+   */
+  bool hasBoundaryCondition(const VariableName & variable, const BoundaryName & boundary) const;
+  /**
+   * Get the name of the boundaries on which the variable should have a boundary condition
+   * @param variable the name of the variable concerned
+   * @param requestor name of the requestor for the boundary condition, used in a potential error
+   * message
+   * @return the name of the boundaries
+   */
+  std::vector<BoundaryName> getBoundaryConditionsBoundaries(const VariableName & variable) const;
+  /**
+   * Get the name of the functor providing the boundary condition for the requested variable and
+   * boundary
+   * @param variable the name of the variable concerned
+   * @param boundary the name of the boundary concerned
+   * @param requestor name of the requestor for the boundary condition, used in a potential error
+   * message
+   * @param bc_type the type of the boundary condition (flux or fixed value)
+   * @return the name of the functor providing the value of the boundary condition
+   */
+  const MooseFunctorName getBoundaryCondition(const VariableName & variable,
+                                              const BoundaryName & boundary,
+                                              const std::string & requestor_name,
+                                              BoundaryConditionType & bc_type) const;
+
+protected:
+  virtual void checkIntegrity() override { checkBoundaryConditionsAllRequested(); }
+
+  /// Names of the variables to set a fixed value BC on
+  const std::vector<VariableName> _fixed_value_bc_variables;
+  /// Names of the variables to set a flux BC on
+  const std::vector<VariableName> _flux_bc_variables;
+
+  // TODO: make our custom string types hashable
+  // This would let us return the boundary condition functor name by reference too
+
+  /// Maps of the fixed value boundary conditions
+  std::unordered_map<std::string, std::unordered_map<std::string, std::string>> _fixed_value_bcs;
+  /// Maps of the flux boundary conditions
+  std::unordered_map<std::string, std::unordered_map<std::string, std::string>> _flux_bcs;
```

Comment:
Hashing strings is O(N). Unless the maps are going to be huge I'd change these to `std::map`

Reply:
changed